### ISSUE TITLE
Accuracy Calculator: when to use the training set as the reference set?

### ISSUE THREAD

**Issue by sarim-zafar** [KevinMusgrave/pytorch-metric-learning#242]
Hi,

In the TripletMarginLossMNIST.ipynb sample notebook, you've used the accuracy calculator in a manner that I don't understand. Your Query and reference embeddings and labels are all the same. The way I understand shouldn't it be like the following:

`accuracies = accuracy_calculator.get_accuracy(test_embeddings, 
                                                train_embeddings,
                                                np.squeeze(test_labels),
                                                np.squeeze(train_labels),
                                                True)`

Assuming that you have train_embeddings and test_embeddings with their respective train_labels and test_labels

**Comment by KevinMusgrave** [comment]
Using ```test_embeddings``` as both the query and reference is valid. It's measuring how well clustered the test set is. For example, consider the case where the test set and training set don't have any overlap of classes. Then using the training set as the reference set would result in an accuracy of 0 for the nearest-neighbor metrics.

If the training set and test set share the same classes (as does MNIST), then your way is valid and more representative of what would happen during deployment:   When a test image comes in, you find the nearest training sample to classify it.

**Comment by KevinMusgrave** [comment]
Maybe I'll change the MNIST examples so that the training set is the reference set. It might be confusing other people the way that it is now.

**Comment by KevinMusgrave** [comment]
I updated the MNIST notebooks
